### PR TITLE
JP-3084: Separate photometric calibration factors for point vs. uniform use

### PR DESCRIPTION
**Pull request by tapastro**
Resolves [JP-3084](https://jira.stsci.edu/browse/JP-3084)

Closes #

This PR addresses an inconsistency in how the scalar conversion factor in the `photom` step is passed through various point vs. extended source conditionals, leading to a 2D conversion array stored in `model.slits[i].photom_uniform` not containing a conversion to MJy/sr. 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by tapastro** [comment]
Regression tests: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/575/

**Review by hbushouse — COMMENTED** [review]
I think the new logic here all looks correct. Of course the PHOTOM_UN array (at least for NIRSpec) still won't have values that will actually convert to surface brightness; they'll just be 1/area, right? I guess we'll have to see if that's good enough to get the master background stuff to work correctly.

**Comment by tapastro** [comment]
After fixing my naming mistake which got the CI tests running, I'm re-running the regression test here: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/577/

**Comment by hbushouse** [comment]
Many of the regression test failures are due to cal files now containing more HDU's than before. Is that expected? Were some extensions, like PHOTOM_UN or PHOTOM_PS, not being included before the fix? Given the difference in HDU's, it's hard to make sense of the comparisons.

**Review by hbushouse — APPROVED** [review]
I think the code all looks correct.

**Comment by tapastro** [comment]
The changes (apart from one which shows the fix including 1/pixelarea_steradians) are outputs that now include the PHOTOM_UN extension which previously did not. I don't know if that is in the "spec" for these outputs, but it seemed like the existing was inconsistent in its decision to save (previously, PHOTOM_UN was saved only for _point-like_ sources), so I think all results are expected and positive changes.

**Comment by hbushouse** [comment]
Doing some archival research into the original implementation of computing and saving both point-source and uniform-source versions of the photometric calibration for NIRSpec (as documented in [JP-1776](https://jira.stsci.edu/browse/JP-1776) and related/child tickets), the only time we need to compute and store both versions is when the slit being processed contains a point source. In that case, both the photom_ps and photom_un arrays are needed later in the master background subtraction step, in order to correct the background data (which are calibrated as uniform source) to match that of a point source. The photom_ps array is not needed for slits that contain an extended/uniform source.

File: jwst/photom/photom.py [hunk 1]
```diff
@@ -795,35 +797,37 @@ def photom_io(self, tabdata, order=None):
...
                                                                    use_wavecorr=True)
                     slit.photom_point = conversion  # store the result
 
                 else:
-                    conversion, no_cal = self.create_2d_conversion(self.input.slits[self.slitnum],
+                    conversion, no_cal = self.create_2d_conversion(slit,
                                                                    self.exptype, conversion,
                                                                    waves, relresps, order)
+
+                    slit.photom_uniform = conversion
```
Comment (hbushouse) [inline jwst/photom/photom.py:829]: Now that I've done the historical research, I don't think this addition is needed. It's in the branch of the if/else that applies if the slit does NOT contain a point source, in which case we don't need to store either `photom_point` or `photom_uniform`. Just apply the computed uniform-source correction to the data, without storing it in the output.

**Review by hbushouse — APPROVED** [review]
I (re)approve.

**Pull request merged**